packages/wrangler/src/__tests__/dev.test.tsx

```diff
@@ -37,6 +37,57 @@ describe("Dev component", () => {
         [32m%s[0m If you think this is a bug then please create an issue at https://github.com/cloudflare/wrangler2/issues/new."
       `);
     });
+
+    it("should use `main` from the top-level environment", async () => {
+      writeWranglerToml({
+        main: "index.js",
+      });
+      fs.writeFileSync("index.js", `export default {};`);
+      await runWrangler("dev");
+      expect((Dev as jest.Mock).mock.calls[0][0].entry.file).toMatch(
+        /index\.js$/
+      );
+      expect(std.out).toMatchInlineSnapshot(`""`);
+      expect(std.warn).toMatchInlineSnapshot(`""`);
+      expect(std.err).toMatchInlineSnapshot(`""`);
+    });
+
+    it("should use `main` from a named environment", async () => {
+      writeWranglerToml({
+        env: {
```

Comment:
let's add a top level `main` here to be sure it never switches

Reply:
That's what the test after this does.

Reply:
My bad, I must've had a brain fart